torchgeo/datasets/treesatai.py

```diff
@@ -0,0 +1,294 @@
+        'Alnus',  # alder
+        'Betula',  # birch
+        'Cleared',  # none
+        'Fagus',  # beech
+        'Fraxinus',  # ash
+        'Larix',  # larch
+        'Picea',  # spruce
+        'Pinus',  # pine
+        'Populus',  # poplar
+        'Prunus',  # cherry
+        'Pseudotsuga',  # Douglas fir
+        'Quercus',  # oak
+        'Tilia',  # linden
+    )
+
+    # https://zenodo.org/records/6780578/files/220629_doc_TreeSatAI_benchmark_archive.pdf
+    all_sensors = ('aerial', 's1', 's2')
+    all_bands: ClassVar[dict[str, list[str]]] = {
+        'aerial': ['IR', 'G', 'B', 'R'],
+        's1': ['VV', 'VH', 'VV/VH'],
+        's2': [
+            'B02',
+            'B03',
+            'B04',
+            'B08',
+            'B05',
+            'B06',
+            'B07',
+            'B8A',
+            'B11',
+            'B12',
+            'B01',
+            'B09',
+        ],
+    }
+    rgb_bands: ClassVar[dict[str, list[str]]] = {
+        'aerial': ['R', 'G', 'B'],
+        's1': ['VV', 'VH', 'VV/VH'],
+        's2': ['B04', 'B03', 'B02'],
+    }
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        split: str = 'train',
+        sensors: Sequence[str] = all_sensors,
+        transforms: Callable[[dict[str, Tensor]], dict[str, Tensor]] | None = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new TreeSatAI instance.
+
+        Args:
+            root: Root directory where dataset can be found.
+            split: Either 'train' or 'test'.
+            sensors: One or more of 'aerial', 's1', and/or 's2'.
+            transforms: A function/transform that takes input sample and its target as
+                entry and returns a transformed version.
+            download: If True, download dataset and store it in the root directory.
+            checksum: If True, check the MD5 of the downloaded files (may be slow).
+
+        Raises:
+            AssertionError: If invalid *sensors* are chosen.
+            DatasetNotFoundError: If dataset is not found and *download* is False.
+        """
+        assert set(sensors) <= set(self.all_sensors)
+
+        self.root = root
+        self.split = split
+        self.sensors = sensors
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+
+        self._verify()
+
+        path = os.path.join(self.root, f'{split}_filenames.lst')
+        with open(path) as f:
+            self.files = f.read().strip().split('\n')
+
+        path = os.path.join(self.root, 'labels', 'TreeSatBA_v9_60m_multi_labels.json')
+        with open(path) as f:
+            self.labels = json.load(f)
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            Length of the dataset.
+        """
+        return len(self.files)
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
```

Comment:
This looks like it returns the multilabel? Shouldn't we support the different tasks possible with the data, so multiclass and multilabel?

Reply:
Multiclass would just be the majority class. I think it's less interesting and unique than multilabel. We can't easily support both without an init parameter.

I would like to add semantic segmentation someday but it requires us to rasterize the GeoJSON files which was more work than I wanted to do on a Sunday.

There are also keypoint detection labels but we don't have a trainer for this and I'm less familiar with how that works.

Reply:
Actually it's a bit more complicated than that. There is:

* 20 species classes (single labels)
* 12 age classes (single labels)
* 15 genus classes (multi labels)

The first two can be found in the filename (and are majority class). The latter is found in the labels JSON file I'm currently using. It isn't super clear in the paper which task they are reporting accuracies for.

Reply:
I'm also not sure whether this is "soft" multilabel classification or multivariate regression. We're basically estimating the percent of an image that corresponds to different classes.

Reply:
Are the labels representing, what fraction of the image is covered by a tree class? or to what confidence degree a tree from that class is present in the image? Intuitively from the labels it would appear that it something like the former, since multilabel confidences do not sum to 1, they would be independent confidences per label right?

Reply:
Yes, the former.

Reply:
Our `MultiLabelClassificationTask` doesn't work because the metrics require binary labels, so I implemented it as a `RegressionTask` for now. Could convert the percentages to binary labels if we want support for multi-label classification.